src/shorthands/buttons.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import statefulSelectors from '../internalHelpers/_statefulSelectors'
+
+const stateMap = [undefined, null, 'active', 'focus', 'hover']
+
+function template(state) {
+  return `button${state},
+  input[type="button"]${state},
+  input[type="reset"]${state},
+  input[type="submit"]${state}`
+}
+
+type State =
+  | typeof(undefined)
+  | null
+  | 'active'
+  | 'focus'
+  | 'hover';
+
+/**
+ * Populates selectors that target all buttons. You can pass optional states to append to the selectors.
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   [buttons('active')]: {
+ *     'border': 'none'
+ *   }
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   ${buttons('active')} {
+ *     border: none;
+ *   }
```

Comment:
This actually doesn't make sense like this in `styled-components`, if anything this should be:

```JS
const div = styled.div`
  > ${buttons('active')} {
    border: none;
  }
`
```